examples/platform/linux/AppMain.cpp

```diff
@@ -427,6 +433,23 @@ CHIP_ERROR InitCommissioner()
     auto & factory = Controller::DeviceControllerFactory::GetInstance();
     ReturnErrorOnFailure(factory.Init(factoryParams));
     ReturnErrorOnFailure(factory.SetupCommissioner(params, gCommissioner));
+
+    chip::FabricInfo * fabricInfo = gCommissioner.GetFabricInfo();
+    VerifyOrReturnError(fabricInfo != nullptr, CHIP_ERROR_INTERNAL);
+
+    uint8_t compressedFabricId[sizeof(uint64_t)] = { 0 };
```

Comment:
It seems like all the impls have pretty much the same bits of logic copied here. Perhaps all of this bit should be shifted into the `DeviceControllerFactory::Init`?

Reply:
This is only available after CommissionerInit, which is done much later than DeviceControllerFactory::Init.

We can restructure the repeated init in a follow-up PR